src/client.rs

```diff
@@ -222,8 +222,7 @@ impl Client {
         if let Some(nodes) = network_nodes.get(most_nodes.0) {
             let pow_feature = String::from("pow");
 
-            // TODO why do we set this in a loop?
-            for (info, node_url) in nodes.iter() {
+            if let Some((info, node_url)) = nodes.first() {
```

Comment:
With that it's missing the part where it inserts it to the synced nodes for the others

Reply:
Yes just noticed it

Reply:
Changed it a bit

Reply:
It still can only add the first node to synced_nodes?

Reply:
This if only updates the protocol parameters with the first node, doesn't deal with synced_nodes

Reply:
The only place where `synced_nodes.insert(` is, is inside of `nodes.first()`
https://github.com/iotaledger/iota.rs/blob/c51a5a7e33d334114f43b5d4e4a8fe61ce1bddf9/src/client.rs#L225-L237

Reply:
I think you are looking at an older version ? This has been changed

Reply:
True, that was the problem